tests/unit/transformers/test_base.py

```diff
@@ -72,3 +436,191 @@ def test__set_columns_data_preserves_order_dataframe(self):
+
+        Setup:
+            - set the ``columns`` attribute to a list of a subset of the columns in the data.
+
+        Input:
+            - a dataframe.
+
+        Expected behavior:
+            - ``column_prefix`` should be set to the elements stored in ``columns`` joined by
+            hashtags (e.g. ['a', 'b'] -> 'a#b').
+            - ``output_columns`` should be set to a list of the keys of what's returned from the
+            ``get_output_types`` method.
+        """
+
+    def test__build_output_columns_generated_already_exist(self):
+        """Test the ``_build_output_columns`` method.
+
+        When this method generates column names that already exist in the data, it should
+        keep adding hashtags at the end of the column name, until the column name becomes unique.
+
+        Setup:
+            - set the ``columns`` attribute to a list of a subset of the columns in the data.
+
+        Input:
+            - a dataframe where the generated column name already exist (e.g. ['a', 'b', 'a#b']).
+
+        Expected behavior:
+            - ``column_prefix`` should be set to the elements stored in ``columns`` joined by
+            hashtags, with a hashtag added at the end (e.g. ['a', 'b'] -> 'a#b#').
+            - ``output_columns`` should be set to a list of the keys of what's returned from the
+            ``get_output_types`` method.
+        """
+
+    def test_fit(self):
+        """Test the ``fit`` method.
+
+        Setup:
+            - mock ``_fit``.
+
+        Input:
+            - a dataframe.
+            - a list of column names from the dataframe.
+
+        Expected behavior:
+            - ``self.columns`` should be set to the passed columns.
+            - ``self._fit`` should be called with the correct data (the subset of the
+            passed dataframe containing only the passed columns).
+            - ``self.output_columns`` should is set correctly.
+            - ``column_prefix`` should be set correctly.
+        """
+
+    def test_transform_incorrect_columns(self):
+        """Test the ``transform`` method when the columns are not in the data.
+
+        When at least on of the passed columns are not present in the data, the method
+        should return the data without doing any transformations.
+
+        Setup:
+            - set ``self.columns`` to a list of columns, where at least one of them is not
+            present in the data
+
+        Input:
+            - a dataframe.
+
+        Output:
+            - the original data.
+        """
+
+    # TODO: should we test all data types being passed? i.e. Series, df and np?
+    def test_transform_drop_false(self):
+        """Test the ``transform`` method when ``drop=False``.
+
+        Setup:
+            - set ``self.columns`` to a list of columns from the data.
+            - set `self.output_columns` to the output column names.
+            - mock self._transform and return some dummy data.
```

Comment:
Same as above, one option is to `patch` the method, but we can also define a dummy transformer that implements a `transform` that simply stores the input data and returns a modified copy of it:

```python3
class Dummy(BaseTransformer):
    def _transform(self, data):
        self._passed_data = data.copy()
        return data.copy() * 2
```